node_exporter.go

```diff
@@ -29,6 +29,35 @@ func init() {
 	prometheus.MustRegister(version.NewCollector("node_exporter"))
 }
 
+func handler(w http.ResponseWriter, r *http.Request) {
+	var filters map[string]bool
+	params := r.URL.Query()["collect[]"]
+	log.Debugln("collect query:", params)
+
+	if len(params) > 0 {
+		filters = make(map[string]bool)
+		for _, param := range params {
+			filters[param] = true
+		}
+	}
+
+	nc, err := collector.NewNodeCollector(filters)
+	if err != nil {
+		log.Fatalf("Couldn't create collector: %s", err)
```

Comment:
We should certainly not exit here if an invalid filter request is given. Return a normal HTTP response error.